Sources/FoundationEssentials/URL/URL.swift

```diff
@@ -2225,11 +2225,40 @@ extension URL {
         #if os(Windows)
         // Convert any "\" to "/" before storing the URL parse info
         var filePath = path.replacing(._backslash, with: ._slash)
+        let isAbsolute: Bool
+        var iter = filePath.utf8.makeIterator()
+        if let driveLetter = iter.next(), driveLetter.isAlpha,
+           iter.next() == ._colon,
+           iter.next() != ._slash {
+            // Drive-relative path: use the current directory for the given drive letter
+            // as the base URL, and remove the drive letter from the relative path.
+            let relativePath = String(Substring(filePath.utf8.dropFirst(2)))
+            let basePath: String? = "\(Unicode.Scalar(driveLetter)):".withCString(encodedAs: UTF16.self) { pwszDriveLetter in
```

Comment:
I wonder if this is better written as:

```swift
let bIsAbsolute: Bool = path.withNTPathRepresentation { !PathIsRelativeW($0) }
```

Reply:
Yeah, that sounds like a much nicer/robust replacement for the `isAbsolute` code, I'll do that, thanks!

I think we would still need to differentiate between `C:relative\path` and `relative\path` if we want to strip the `C:` before storing the relative path in `URL`, so we'll still need to do the alpha/colon/slash check here.

Reply:
Hm, I'm seeing `PathIsRelativeW` return `FALSE` for `C:relative\path` (probably because it's resolved against the absolute cwd for `C:`), so it might not work for this use case.

Reply:
We could rework the `URL(filePath:)` initializer to immediately standardize `C:relative\path` to its absolute representation, and not use the relative path + base URL distinction. This would essentially be:

```
bIsAbsolute = !PathIsRelativeW(path)
if bIsAbsolute {
  path = GetFullPathNameW(path)
  baseURL = nil
} else {
  // path is relative
  baseURL = URL.currentDirectory()
}
```

`PathIsRelative("/absolute/path")` does return `TRUE`, though, so we'd need to make sure we strip the leading slash in other code when merging with the base path.